0.download_data/scripts/1.view_the data_structure.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# This notebook looks at the downloaded mnist data and looks into the image dataloader and rolling window data class.
+# Noteably this code is modified from the original code written by original code by Xinqiang Ding <xqding@umich.edu>
+
+# In[1]:
+
+
+import pathlib
+import sys
+
+# mnist data parser import
+import mnist
+import numpy as np
+import pandas as pd
+
+sys.path.append("../../utils")
+
+import pickle
+
+import matplotlib.pyplot as plt
+from prep_data import MNIST_Dataset
+from torch.utils.data import DataLoader
+
+# In[2]:
+
+
+# open the data
+# set path to the data
+mnist_pickle_path = pathlib.Path("../../data/mnist/MNIST.pkl").resolve(strict=True)
+
+with open(mnist_pickle_path, "rb") as file_handle:
+    MNIST = pickle.load(file_handle)
+
+# In[3]:
+
+
+# get the MNIST data keys
+print(MNIST.keys())
+# MNIST["train_label"][:15]
+MNIST["train_image"][0].shape
+
+# In[4]:
+
+
+# set the batch size
+batch_size = 8
+
+# create the data class
+# this class makes a rolling window of the data
+data = MNIST_Dataset(
+    MNIST["train_image"], MNIST["train_label"], binary=False, number_of_frames=10
+)
+# create the data loader
+data_loader = DataLoader(data, batch_size=batch_size, shuffle=True)
+# get the first batch of data
+batch, sample = next(enumerate(data_loader))
+
+# In[5]:
+
+
+idx = sample[0]
+images = sample[1]["image"]
+labels = sample[1]["label"]
+print(f"{len(idx)} images in the batch")
+print(f"labels: {labels.shape}")
+print(f"images: {images.shape}")
+
+# In[6]:
+
+
+# number of images to vizualize
```

Comment:
I am not sure what this references, recommend removing or making more explicit.

Reply:
Removed old comment!